lib/Component.php

```diff
@@ -16,9 +16,14 @@ public function __construct($factory, $component_data) {
     $this->spec     = $component_data['spec'];
     $this->name     = $this->spec->name;
     $this->version  = $this->spec->version;
-    $this->location = $component_data['location'];
-    $this->subpath  = $component_data['subpath'];
-    $this->type     = isset($this->spec->type) ? $this->spec->type : 'backend';
+
+    if (array_key_exists('base_dir', $component_data)) {
```

Comment:
It is probably a good idea to set both $this->base_dir and $this->base_url to some default value even if the component data does not provide them. I am perfectly happy with using **null**, the important thing is that the member variables is always defined.